crates/bevy_ecs/src/component.rs

```diff
@@ -1834,3 +1926,106 @@ impl RequiredComponents {
         }
     }
 }
+
+/// Component [clone handler function](ComponentCloneFn) implemented using the [`Clone`] trait.
+/// Can be [set](ComponentCloneHandlers::set_component_handler) as clone handler for the specific component it is implemented for.
+/// It will panic if set as handler for any other component.
+///
+/// See [`ComponentCloneHandlers`] for more details
+pub fn component_clone_via_clone<C: Clone + Component>(
+    world: &mut World,
+    _component_id: ComponentId,
+    entity_cloner: &EntityCloner,
+) {
+    let component = world
+        .entity(entity_cloner.get_source())
+        .get::<C>()
+        .expect("Component must exists on source entity")
+        .clone();
+    world
+        .entity_mut(entity_cloner.get_target())
+        .insert(component);
+}
+
+/// Component [clone handler function](ComponentCloneFn) implemented using reflect.
+/// Can be [set](ComponentCloneHandlers::set_component_handler) as clone handler for any registered component,
+/// but only reflected components will be cloned.
+///
+/// See [`ComponentCloneHandlers`] for more details
+#[cfg(feature = "bevy_reflect")]
+pub fn component_clone_via_reflect(
+    world: &mut World,
+    component_id: ComponentId,
+    entity_cloner: &EntityCloner,
+) {
+    world.resource_scope::<crate::reflect::AppTypeRegistry, ()>(|world, registry| {
+        let registry = registry.read();
+
+        let component_info = world
+            .components()
+            .get_info(component_id)
+            .expect("Component must be registered");
+        let Some(type_id) = component_info.type_id() else {
+            return;
+        };
+        let Some(reflect_component) =
+            registry.get_type_data::<crate::reflect::ReflectComponent>(type_id)
+        else {
+            return;
+        };
+        let source_component = reflect_component
+            .reflect(
+                world
+                    .get_entity(entity_cloner.get_source())
+                    .expect("Source entity must exist"),
+            )
+            .expect("Source entity must have reflected component")
+            .clone_value();
+        let mut target = world
+            .get_entity_mut(entity_cloner.get_target())
+            .expect("Target entity must exist");
+        reflect_component.apply_or_insert(&mut target, &*source_component, &registry);
+    });
+}
+
+/// Noop implementation of component clone handler function
+///
+/// See [`ComponentCloneHandlers`] for more details
+pub fn component_clone_ignore(
+    _world: &mut World,
+    _component_id: ComponentId,
+    _entity_cloner: &EntityCloner,
+) {
+}
+
+/// Wrapper for components clone specialization using autoderef
```

Comment:
That's true! Didn't think about that because generic components in general are a bit of a pain to work with. Not really sure where to add a note about that though.

Reply:
Maybe there should be a section in the doc comments for `EntityCloneBuilder` that explain the strategy for choosing the clone handler.  There are actually quite a few ways it can be set: `EntityCloneBuilder::override_component_clone_handler()`, `Component::get_component_clone_handler()`, `ComponentCloneHandlers::set_component_handler()`, `ComponentCloneHandlers::set_default_handler()`, and the `bevy_reflect` feature flag.  And the docs for those methods could link back to that section so you can see their context.